reference/nonfungible-token.scilla

```diff
@@ -64,257 +71,472 @@ let makeErrorEvent =
+(* @param:  minter      - Address of the minter to be approved or removed            *)
+(* Returns error event CodeNotAuthorised if _sender is not contract_owner.           *)
+transition configureMinter(minter: ByStr20)
+  is_owner = builtin eq contract_owner _sender;
+  match is_owner with
+  | False =>
+    (* Token exists, return error code *)
+    err = CodeNotAuthorised;
+    EmitError err
+  | True =>
+    some_minter <- minters[minter];
+    match some_minter with
+    | Some Unit => 
+      delete minters[minter];
+      (* Emit success event *)
+      e = {_eventname: "RemovedMinterSuccess"; minter: minter};
+      event e
+    | None =>
+      authorize = Unit;
+      minters[minter] := authorize;
+      (* Emit success event *)
+      e = {_eventname: "AddMinterSuccess"; minter: minter};
+      event e
+    end
+  end
+end
+
+(* @dev:    Mint new tokens. Only contract_owner can mint.         *)
+(* @param:  to        - Address of the token recipient             *)
+(* @param:  token_id  - ID of the new token to be minted           *)
+(* @param:  token_uri -  URI of the the new token to be minted     *)
+(* Returns error event CodeTokenExists if token already exists.    *)
+(* Returns error event CodeNotAuthorised if _sender is not minter. *)
+(* Revert transition if invalid recipient contract.                *)
+transition mint(to: ByStr20, token_id: Uint256, token_uri: String)
   (* Check if token exists *)
-  tokenExist <- exists tokenOwners[tokenId];
-  match tokenExist with
+  token_exist <- exists token_owners[token_id];
+  match token_exist with
   | True =>
     (* Token exists, return error code *)
     err = CodeTokenExists;
-    MakeError err
+    EmitError err
   | False =>
-    (* Check if sender is contractOwner *)
-    isContractOwner = builtin eq _sender contractOwner;
-    match isContractOwner with
+    (* Check if sender is contract_owner *)
+    is_contract_owner = builtin eq _sender contract_owner;
+    (* Check if sender is a minter *)
+    is_minter <- exists minters[_sender];
+    is_authorised = orb is_contract_owner is_minter;
+    match is_authorised with
     | True =>
       (* Mint new token *)
-      tokenOwners[tokenId] := to;
+      token_owners[token_id] := to;
       (* Add to owner's count *)
-      someCurrentCount <- ownedTokenCount[to];
-      newCount = addOwnerCount someCurrentCount;
-      ownedTokenCount[to] := newCount;
+      some_current_count <- owned_token_count[to];
+      new_count = add_owner_count some_current_count;
+      owned_token_count[to] := new_count;
+      (* Add token_uri for token_id *)
+      token_uris[token_id] := token_uri;
+      (* Add to total_supply *)
+      current_supply <- total_supply;
+      new_supply = builtin add current_supply one;
+      total_supply := new_supply; 
       (* Emit success event *)
-      e = {_eventname: "MintSuccess"; by: _sender; recipient: to; token: tokenId};
+      e = {_eventname: "MintSuccess"; by: _sender; recipient: to;
```

Comment:
I think we have to standardise on the case for `_eventname`. Some of them are camelCase but some are PascalCase

Reply:
I believe that all `_eventname` are currently in PascalCase.